frontend/src/pages/UserDashboard/UserDashboard.jsx

```diff
@@ -0,0 +1,131 @@
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {  fetchEnrolledCourses, fetchUserProfile, fetchNotEnrolledCourses } from '../../services/api';
+import styles from './UserDashboard.module.css';
+
+const UserDashboard = () => {
+  const [activeTab, setActiveTab] = useState('Home');
+  const [profileData, setProfileData] = useState(null);
+  const [coursesData, setCoursesData] = useState([]);
+  const [enrolledCoursesData, setEnrolledCoursesData] = useState([]);
+  const userId = localStorage.getItem('userId'); 
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        if (activeTab === 'Profile') {
```

Comment:
These should probably be routes, not just tabs, so that when the user refreshes they stay on the same page. Also, don't use string literals for stuff like this - someone at some point will accidentally type `"profile"` instead of `"Profile"` or `"My courses"` instead of `"My Courses"` and a part of the app will break silently.